### ISSUE TITLE
false positive for no-useless-escape

### ISSUE THREAD

**Issue by blord-fullscreen** [eslint/eslint#8524]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 6.10.2
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**What did you do? Please include the actual source code causing the issue.**
```
    if (JSON.stringify(data).match(/data:[\w\/+]+;base64,/gi)) {
      const errorData = { error: 'Please do not upload base 64'};
      return Promise.reject(errorData);
    }
```

**What did you expect to happen?**

The regex should be ok.

**What actually happened? Please include the actual, raw output from ESLint.**
```
error  Unnecessary escape character: \/  no-useless-escape
```
The `no-useless-escape` rule is flagging the escaped slash between the word character and the plus symbol. I think this is incorrect. The escape isn't 'useless'. Playing around with the RegEx at <a href="https://regex101.com/">regex101.com</a>, I don't see a way to rewrite it without escaping the slash.

Just to provide practical context, this code is to detect strings like <a href="https://jsfiddle.net/casiano/xadvz/">this</a>. Thanks for looking.

**Comment by platinumazure** [comment]
Thanks for the issue!

In JavaScript, a slash inside a character class doesn't need to be escaped, so that escape is indeed useless. So I believe this is working as intended.

**Comment by blord-fullscreen** [comment]
Thanks @platinumazure, you're right. I was mistakenly testing RegEx101 in its PHP flavor, and not in the JS flavor.
